pkg/controllers/securitypolicy_associated_controller.go

```diff
@@ -0,0 +1,241 @@
+	"context"
+	"reflect"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+)
+
+var ignoreNs = []string{"kube-", "vmware-"}
+
+// We should consider the below sceneries:
+// When a new added pod whose port name exists in security policy.
+// When a deleted pod whose port name exists in security policy.
+// When a pod's label is changed.
+// When a namespace's label is changed.
+// In summary, we could roughly think if the port name of security policy exists in the
+// new pod or old pod, we should reconcile the security policy.
+
+type EnqueueRequestForNamespace struct {
+	Client client.Client
+}
+
+func (e *EnqueueRequestForNamespace) Create(_ event.CreateEvent, _ workqueue.RateLimitingInterface) {
+	log.V(1).Info("namespace create event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Delete(_ event.DeleteEvent, _ workqueue.RateLimitingInterface) {
+	log.V(1).Info("namespace delete event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Generic(_ event.GenericEvent, _ workqueue.RateLimitingInterface) {
+	log.V(1).Info("namespace generic event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Update(updateEvent event.UpdateEvent, l workqueue.RateLimitingInterface) {
+	obj := updateEvent.ObjectNew.(*v1.Namespace)
+
+	podList := &v1.PodList{}
+	err := e.Client.List(context.Background(), podList, client.InNamespace(obj.Name))
+	if err != nil {
+		log.Error(err, "failed to list pod in namespace", "namespace", obj.Name)
+		return
+	}
+
+	shouldReconcile := false
+	for _, pod := range podList.Items {
+		if checkPodHasNamedPort(pod, "update") {
+			shouldReconcile = true
+			break
+		}
+	}
+	if !shouldReconcile {
+		log.Info("no pod in namespace is relevant", "namespace", obj.Namespace, "name", obj.Name)
+		return
+	}
+
+	err = reconcileSecurityPolicy(e.Client, podList.Items, l)
+	if err != nil {
+		log.Error(err, "Failed to reconcile security policy")
+	}
+}
+
+type EnqueueRequestForPod struct {
+	Client client.Client
+}
+
+func (e *EnqueueRequestForPod) Create(createEvent event.CreateEvent, l workqueue.RateLimitingInterface) {
+	e.Raw(createEvent, l)
+}
+
+func (e *EnqueueRequestForPod) Update(updateEvent event.UpdateEvent, l workqueue.RateLimitingInterface) {
+	e.Raw(updateEvent, l)
+}
+
+func (e *EnqueueRequestForPod) Delete(deleteEvent event.DeleteEvent, l workqueue.RateLimitingInterface) {
+	e.Raw(deleteEvent, l)
+}
+
+func (e *EnqueueRequestForPod) Generic(genericEvent event.GenericEvent, l workqueue.RateLimitingInterface) {
+	e.Raw(genericEvent, l)
+}
+
+func (e *EnqueueRequestForPod) Raw(evt interface{}, q workqueue.RateLimitingInterface) {
```

Comment:
Why in some places you named the instance of `workqueue.RateLimitingInterface` to `l` (e.g. L94) and in the other places it's `q` (e.g. L98)? Could you please unify the var name for the same struct?

Reply:
ok